src/promptflow-evals/promptflow/evals/evaluators/_groundedness/flow/evaluate_with_rai_service.py

```diff
@@ -0,0 +1,143 @@
+import importlib.metadata
+import json
+import time
+from typing import List
+from urllib.parse import urlparse
+
+import requests
+from azure.core.credentials import TokenCredential
+from azure.identity import DefaultAzureCredential
+from constants import EvaluationMetrics, RAIService, Tasks
+
+from promptflow.core import tool
+
+try:
+    version = importlib.metadata.version("promptflow-evals")
+except importlib.metadata.PackageNotFoundError:
+    version = "unknown"
+USER_AGENT = "{}/{}".format("promptflow-evals", version)
+
+
+def ensure_service_availability(rai_svc_url: str):
+    svc_liveness_url = rai_svc_url.split("/subscriptions")[0] + "/meta/version"
+    response = requests.get(svc_liveness_url)
+    if response.status_code != 200:
+        raise Exception("RAI service is not available in this region")
+
+
+def submit_request(question: str, answer: str, context: str, rai_svc_url: str, credential: TokenCredential):
+    user_text = {
+        "question": question,
+        "answer": answer,
+        "context": context,
+    }
+    user_text_json = json.dumps(user_text)
+    payload = {
+        "UserTextList": [user_text_json],
+        "AnnotationTask": Tasks.GROUNDEDNESS,
+        "MetricList": [EvaluationMetrics.GROUNDEDNESS],
+    }
```

Comment:
groundedness is only available in eastus2 and swedencentral. You could call /checkannotation to see if groundedness is available in a given region and then only submit annotation request.

Reply:
Got it. Thanks for the information. We will add this logic post build.

Reply:
ADO: [Task 3028927](https://msdata.visualstudio.com/Vienna/_workitems/edit/3028927): Switch to use new way to check RAI service availability